### DISCUSSION TITLE
Why is nextjs app reloading when I hit API endpoint?

### DISCUSSION THREAD

**Discussion by Hemanth-M-K** [vercel/next.js#33172] [Help]
Hello, 

I have a nextjs app that is running with two pages:

/demo
/api/signup
For some reason when I first hit the /api/signup endpoint (or simply refresh the page) it will trigger a HMR reload of the app. This also happens when making fetch calls to the api, the app reload's the the request made to the API does not work because of the interruption. Why is this happening?

Subsequent calls to the api will not result in a reload.

This happens when using next dev and next build && next start.

you can check more information in stackover flow (https://stackoverflow.com/questions/70498603/why-is-nextjs-app-reloading-when-i-hit-api-endpoint)

**Comment by tzt9597pg** [comment]
wondering

**Comment by cupofjoakim** [comment]
Just running in to the same issue 3 years later. @Hemanth-M-K did you end up figuring out the issue?

**Comment by jorgvm** [comment]
In my case, the issue was caused by Next.js lazily compiling routes during development. The first time an API route was requested, it triggered a regeneration of the central _buildManifest.js. In a large application, this file can become so large that webpack cannot apply the update incrementally on the client. As a result, Fast Refresh falls back to performing a full page reload.

Switching to Turbopack resolved the issue for me, as it can handle these updates properly.

**Comment by cupofjoakim** [comment]
I can update on our end too. Our issue was also the lazy compilation of the routes during development. For us a turbo migration is not in the cards (yet) due to some odd webpack dependencies. 

We worked around the issue by adding a call to the api endpoints on server startup to trigger the compilation early. Not great, but we do it only in dev and it gets us there.
